packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -569,6 +499,172 @@ export function getValidatorApi(
     }
   }
 
+  async function produceBuilderBlockBody(
+    // as of now fee recipient checks can not be performed because builder does not return bid recipient
+    opts: {
+      slot: Slot;
+      version: ForkName;
+      randaoReveal: BLSSignature;
+      graffiti: Buffer;
+      parentBlockRoot: Root;
+      currentState: CachedBeaconStateAllForks;
+    }
+  ): Promise<AssembledBlockBodyResponse<BlockType.Blinded>> {
+    const {slot, version} = opts;
+    if (!isForkPostBellatrix(version)) {
+      throw Error(`Invalid fork=${version} for produceBuilderBlindedBlockBody`);
+    }
+
+    const source = ProducedBlockSource.builder;
+    metrics?.blockProductionRequests.inc({source});
+
+    // Error early for builder if builder flow not active
+    if (!chain.executionBuilder) {
+      throw Error("Execution builder not set");
+    }
+    if (!chain.executionBuilder.status) {
+      throw Error("Execution builder disabled");
+    }
+
+    const resp = await chain.produceBlindedBlockBody(opts);
+
+    metrics?.blockProductionSuccess.inc({source});
+    logger.verbose("Produced blinded block body", {
```

Comment:
the old flow has `loggerContext` so need to have that detailed info for debugging purpose later
same for other functions